proposals/data-classes.md

```diff
@@ -0,0 +1,241 @@
+
+# Working with Data
+
+When we talk about C# and start to talk about data, the conversation often
+moves to talk about advanced data structures and complicated use cases. Here
+I'd like to do the opposite: talk about simple data and the representations
+we use for it.
+
+To start, let's talk about what data is and what it isn't.
+
+Data *is* a collection of values with potentially heterogenous types. Some
+examples include
+
+* Database rows
+* JSON/XML messages
+* Login info
+* Configuration options
+
+Data *is not*
+
+* A process
+* A computation
+* A conversation
+* Interactive
+* An object
```

Comment:
This example I'm unclear about.

Inheritance is explicitly mentioned further down, so what makes it _not_ an object? Lack of a complex graph?

Reply:
Inheritance can be possible without changing the contract with the type, namely that it is simple data and does not represent an interactive process.